go/viperutil/internal/sync/sync_test.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sync_test
```

Comment:
```suggestion
package sync
```

Reply:
this is deliberate. it forces me to write the tests without reaching into `package sync`'s internals; i _have_ to use the publicly-exported API